src/topic0/abstractFactory/factory/FactoryProducer.java

```diff
@@ -0,0 +1,17 @@
+package topic0.abstractFactory.factory;
+
+public class FactoryProducer {
+
+    public static AbstractSQLFactory getFactory(String sqlType){
+        if (sqlType.equalsIgnoreCase("MySQL")){
```

Comment:
It would be better to use [yoda conditions](https://knowthecode.io/yoda-conditions-yoda-not-yoda)